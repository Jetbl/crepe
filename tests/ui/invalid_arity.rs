// In this test, the relation 'Fib' is given the wrong number of arguments.

mod datalog {
    use crepe::crepe;

    crepe! {
        @output
        struct Fib(u32, u32);

        Fib(0, 0, 2);
        Fib(1, 1);
    }
}

fn main() {}
