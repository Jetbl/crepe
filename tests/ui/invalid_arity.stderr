error: Relation 'Fib' was declared with arity 2, but constructed with arity 3 here.
  --> $DIR/invalid_arity.rs:10:9
   |
10 |         Fib(0, 0, 2) <- (true);
   |         ^^^
